src/Simulation/CsharpGeneration/Resources/EntryPointDriver.cs

```diff
@@ -0,0 +1,355 @@
+            }
+            return value;
+        };
+
+        /// <summary>
+        /// Parses a Q# range from an enumerable of strings, where the items are start and end or start, step, and end.
+        /// </summary>
+        /// <param name="option">The name of the option being parsed.</param>
+        /// <param name="arg">The argument string for the option.</param>
+        /// <param name="items">The items in the argument.</param>
+        /// <returns>The result of parsing the strings.</returns>
+        private static @Result<QRange> ParseRangeFromEnumerable(string option, string arg, IEnumerable<string> items) =>
+            items.Select(item => TryParseLong(option, item)).Sequence().Bind(values =>
+                values.Count() == 2
+                ? @Result<QRange>.Success(new QRange(values.ElementAt(0), values.ElementAt(1)))
+                : values.Count() == 3
+                ? @Result<QRange>.Success(new QRange(values.ElementAt(0), values.ElementAt(1), values.ElementAt(2)))
+                : @Result<QRange>.Failure(GetArgumentErrorMessage(option, arg, typeof(QRange))));
+
+        /// <summary>
+        /// Parses a long from a string.
+        /// </summary>
+        /// <param name="option">The name of the option being parsed.</param>
+        /// <param name="str">The string to parse.</param>
+        /// <returns>The result of parsing the string.</returns>
+        private static @Result<long> TryParseLong(string option, string str) =>
+            long.TryParse(str, out var result)
+            ? @Result<long>.Success(result)
+            : @Result<long>.Failure(GetArgumentErrorMessage(option, str, typeof(long)));
+
+        /// <summary>
+        /// Returns an error message string for an argument parser.
+        /// </summary>
+        /// <param name="option">The name of the option.</param>
+        /// <param name="arg">The value of the argument being parsed.</param>
+        /// <param name="type">The expected type of the argument.</param>
+        /// <returns></returns>
+        private static string GetArgumentErrorMessage(string option, string arg, Type type) =>
+            $"Cannot parse argument '{arg}' for option '{option}' as expected type {type}.";
+
+        /// <summary>
+        /// Displays an error message for using a non-default custom simulator.
+        /// </summary>
+        /// <param name="name">The name of the custom simulator.</param>
+        private static void DisplayCustomSimulatorError(string name)
+        {
+            var originalForeground = Console.ForegroundColor;
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.Error.WriteLine($"The simulator '{name}' could not be found.");
+            Console.ForegroundColor = originalForeground;
+            Console.Error.WriteLine();
+            Console.Error.WriteLine(
+                $"If '{name}' is a custom simulator, it must be set in the DefaultSimulator project property:");
```

Comment:
I take it you went with disallowing defining execution on a custom simulator via command line? (works for me, just double checking)

Reply:
Yes. I did not want to go to the trouble of updating the reflection loading to use a separate option for the assembly name. :)